x-pack/plugins/observability/public/pages/alert_details/components/alert_summary.test.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import * as useUiSettingHook from '@kbn/kibana-react-plugin/public/ui_settings/use_ui_setting';
+import { render } from '../../../utils/test_helper';
+import { AlertSummary } from './alert_summary';
+import { asDuration } from '../../../../common/utils/formatters';
+import { kibanaStartMock } from '../../../utils/kibana_react.mock';
+import { useKibana } from '../../../utils/kibana_react';
+import { triggersActionsUiMock } from '@kbn/triggers-actions-ui-plugin/public/mocks';
+import { waitFor } from '@testing-library/react';
+import { alertWithTags, alertWithNoData, tags } from '../mock';
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useParams: jest.fn(),
+}));
+
+jest.mock('../../../utils/kibana_react');
+
+const useKibanaMock = useKibana as jest.Mock;
+
+const mockKibana = () => {
+  useKibanaMock.mockReturnValue({
+    services: {
+      ...kibanaStartMock.startContract(),
+      triggersActionsUi: triggersActionsUiMock.createStart(),
+    },
+  });
+};
+
+describe('Alert summary', () => {
+  jest
+    .spyOn(useUiSettingHook, 'useUiSetting')
+    .mockImplementation(() => 'MMM D, YYYY @ HH:mm:ss.SSS');
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockKibana();
+  });
+
+  it('should show alert data', async () => {
+    const alertSummary = render(<AlertSummary alert={alertWithTags} />);
+
+    expect(alertSummary.queryByText('1957')).toBeInTheDocument();
+    expect(alertSummary.queryByText(asDuration(882076000))).toBeInTheDocument();
+    expect(alertSummary.queryByText('Active')).toBeInTheDocument();
+    expect(alertSummary.queryByText('Sep 2, 2021 @ 08:54:09.674')).toBeInTheDocument();
+    expect(
+      alertSummary.getByText('Sep 2, 2021 @ 09:08:51.750', { exact: false })
+    ).toBeInTheDocument();
+    await waitFor(() => expect(alertSummary.queryByTestId('tagsOutPopover')).toBeInTheDocument());
```

Comment:
Just out of curiosity: why do you need to wait for this one?

Reply:
It was detecting only spinner in place of tags without waiting.